### ISSUE TITLE
http.Http's handling of responses that have no explicit length is broken

### ISSUE THREAD

**Issue by ousado** [HaxeFoundation/haxe#5759]
TL;DR it's possible that under certain conditions reading from a socket returns -1 with errno set to ECONNRESET (Connection reset by peer) which depending on the application-level may or may not be an error but there's no way to distinguish that from other errors, neither using the Haxe networking API, let alone using haxe.Http  

It can (and usually should) be an application-level error, but not in all cases. HTTP without explicit length (meaning neither chunked transfer-encoding is used nor an explicit Content-Length is provided) allows this, for instance - the client simply is supposed to read until the connection is closed. Under certain circumstances - the client already shut down the connection and the server closed the connection, too, it's possible that instead of returning 0 (which signals the connection was closed) read/recv/etc. return -1 with ECONNRESET, and that's not actually an error in this case.

**Comment by ousado** [comment]
Another issue that can surface in this context is ENOTCONN on shutdown (neko throws "std@socket_shutdown").
This happens when the connection has been closed by the server before the shutdown in line https://github.com/HaxeFoundation/haxe/blob/development/std/haxe/Http.hx#L700 is done.
The system call fails with:
shutdown(3, SHUT_WR)                    = -1 ENOTCONN (Transport endpoint is not connected)
and again, the data is lost in the current haxe.Http implementation.
In this case the error isn't even translated into a haxe-representation of haxe.io.Error. However the data could still be read from the socket - just because it's not connected anymore, that doesn't mean the RX buffer has been drained. 
This can be avoided by simply catching the error.

**Comment by thomasuster** [comment]
My PR handles one of the use cases you mentioned at C level for neko platform, which may or may not be good, but anyone interested in issue should know it's related. https://github.com/HaxeFoundation/neko/pull/150
